### PR TITLE
Update github actions.

### PR DESCRIPTION
**Pull request by neetopia**
* Do not run tests on auto-merge.
* Run daily tests for release branches.
* Test for all platforms in release action.

### CODE REVIEW

**Comment by neetopia** [comment]
triggered actions: https://github.com/neetopia/ksp/actions

auto-merge is failing due to conflicts at the moment.

**Comment by neetopia** [comment]
might need to wait til 4:30 am to verify schedule working.

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+          key: ${{ runner.os }}-gradle-wrapper-${{ hashFiles('gradle/wrapper/gradle-wrapper.properties') }}
+      # Run ktlint
+      - name: lint
+        if: matrix.os == 'ubuntu-latest'
+        run: ./gradlew ktlint
+
+      # Check API compatibility
+      - name: API compatibility check
+        if: matrix.os == 'ubuntu-latest'
+        run: ./gradlew :api:checkApi
```
Comment (ting-yuan) [inline .github/workflows/nightly.yml:57]: Let's remove API check and lint. API check is performed in pushes and auto merges. Lint isn't relevant on release branches and could be broken by auto-merge.

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        run: echo ::set-env name=JDK_9::$(echo $JAVA_HOME)
+      - name: Setup Java 11
+        uses: actions/setup-java@v1.4.3
+        with:
+          java-version: '11'
+          java-package: jdk
+          architecture: x64
+
+      # Checkout
+      - uses: actions/checkout@v2
```
Comment (ting-yuan) [inline .github/workflows/nightly.yml:33]: This tests all three OS on the main branch. There are 9 combinations though:
{main, next-release, current-release} x {windows, mac, linux}

where {main, current-release} x {linux, mac} are tested on pushes (main.yml) already. Can you make this a matrix and skip {main, current-release} x {linux, mac}?

**Review by ting-yuan — COMMENTED** [review]
Another thought: what if we keep tests on linux and mac in auto-merge and only run test on windows nightly? so that
1. every branch is tested on linux and mac upon push / auto-merge.
2. nightly and release only test on windows.

For reference, there are some overlaps and misses in this PR:
1. main.yml / on-push test {main, current-release} x {linux, mac}
2. nightly tests {main} x {linux, mac, windows}
3. release tests {current-release, next-release} x {linux, mac, windows}

**Comment by neetopia** [comment]
Closing this one in favor of #947

**Pull request closed without merging**